docs/DeveloperGuide.md

```diff
@@ -144,92 +139,88 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+
+##### Design choice
+In the context of tag management, this design choice is aimed at properly encapsulating two separate functionalities.
+1. `AddressBook` - uses `TagManager` to track which `Person` objects falls under which `Tag`s. 
+    - `TagManager` is necessary to avoid cyclic dependency between `Person` and `Tag`. 
+    - `TagManager` is embedded in `AddressBook` so whenever there is a change to the list of `Person`s, the tag-to-person map can be updated immediately.
+2. `TagTree` tracks the tag-to-tag relationships only. In particular, it keeps track of which set of `Tag`s are child-tags of which `Tag`. 
+
+Using these two mutable constructs, it allows for accurate realtime queries by higher-level components even though the internal mapping changes frequently between commands.
+In order to ensure that the right commands are called at the right time, `Model` implements only a limited set of methods to be used that can change the internal mapping.
+To support a greater varity of `Command`s, ensure that the correct methods from either of `AddressBook`, `TagTree` or `ContactTagIntegrationManager` is chosen. 
+A rule of thumb is to search for the method in `ContactTagIntegrationManager` first before looking for a similar method in the other two classes. 
+ 
+To view the full list of methods and documentation for the three major classes, you can view them at [`AddressBook`](https://github.com/AY2021S1-CS2103T-W10-4/tp/blob/master/src/main/java/seedu/address/model/AddressBook.java), 
+[`TagTree`](https://github.com/AY2021S1-CS2103T-W10-4/tp/blob/master/src/main/java/seedu/address/model/tag/TagTree.java) and [`ContactTagIntegrationManager`](https://github.com/AY2021S1-CS2103T-W10-4/tp/blob/master/src/main/java/seedu/address/model/ContactTagIntegrationManager.java).
+
+##### Design improvement
+A possible way to improve the current design is to remove dependency from `ModelManager` to `TagTree`, `AddressBook` and `ContactTagIntegrationManager` by creating another Facade class containing these three classes.
+As such, only the relevant methods required for specific `Command`s to work will be exposed to `Model`. For example, right now, both `ContactTagIntegrationManager`'s `deleteTag(Tag)` method and `TagTree`'s `deleteTag(Tag)` method are exposed to `ModelManager` when only one of them is actually used.
+This makes it easier for others working at a similar level of abstraction to avoid using the wrong methods.  
+
+##### Additional Notes:
+_Definitions_:
+- For a `Tag` to _exist_ in Athena, the `Tag` must have either at least one `Person` with the `Tag` **or** at least one child-tag.
+- A _child-tag_ of a tag signifies a directional relation from a tag to its _child-tag_. It allows for some commands that affect a tag to also affect its _child-tag_. The reverse cannot be done. The other tag in the relationship is the _parent-tag_.
+- _Parent-tag_: see _child-tag_.
```

Comment:
is this necessary

Reply:
I believe so. This is for when someone performs ctrl+f "parent-tag". I could restate the same thing for "parent-tag" but I think the two are close enough in proximity unlike in the glossary in UG.

Reply:
okay!